d2l/train.py

```diff
@@ -0,0 +1,37 @@
+
+import torch
+import torch.optim as optim
+
+
+def evaluate_accuracy(data_iter, net):
+    """Evaluate accuracy of a model on the given data set."""
+    acc_sum, n = torch.Tensor([0]), 0
+    for batch in data_iter:
+        features, labels = batch
+        for X, y in zip(features, labels):
+            y = y.type(torch.FloatTensor)
+            acc_sum += torch.sum((torch.argmax(net(X), dim=1).type(torch.FloatTensor) == y).data).float()
```

Comment:
I think It is a better practice to use `.item` instead of `.data` in this context in PyTorch.
`.data` will be deprecated in the newer versions. 
Please make the changes replacing .item in all the following lines.

Reply:
`.item()` can only be used to convert single element tensors to scalars. So, I've used `.detach()` for the multi-element cases.

Reply:
Yeah!! The loss will be a scalar always. Looks good now.